cpp/src/arrow/flight/sql/client.h

```diff
@@ -329,6 +329,25 @@ class ARROW_FLIGHT_SQL_EXPORT FlightSqlClient {
   /// \param[in] info         The FlightInfo of the query to cancel.
   ::arrow::Result<CancelResult> CancelQuery(const FlightCallOptions& options,
                                             const FlightInfo& info);
+   
+  /// \brief Sets session options.
+  ///
+  /// \param[in] options            RPC-layer hints for this call.
+  /// \param[in] session_options    The session options to set.
+  ::arrow::Result<std::vector<SetSessionOptionResult>> SetSessionOptions(
```

Comment:
I think we need a may to set SessionOptionResult in a GetFlightInfo call as the likes of Power BI won't handle two calls from a driver easily.

Reply:
SetSessionOptions will be part of the ODBC driver connection init, after which cookies (presumably persisted by the ODBC driver which it might need to be updated to support, or by the ODBC-using application which I don't think is the case(?!)) will persist the session state reference.